packages/legend-graph/src/__tests__/roundtripTestData/TEST_DATA__DomainRoundtrip.ts

```diff
@@ -947,3 +947,48 @@ export const TEST_DATA__MeasureRoundtrip = [
     classifierPath: 'meta::pure::metamodel::type::Measure',
   },
 ];
+
+export const TEST_DATA__VersionedClassRoundtrip = [
+  {
+    path: '__internal__::SectionIndex',
+    content: {
+      _type: 'sectionIndex',
+      name: 'SectionIndex',
+      package: '__internal__',
+      sections: [
+        {
+          _type: 'importAware',
+          elements: ['test::C'],
+          imports: ['test'],
+          parserName: 'Pure',
+        },
+      ],
+    },
+    classifierPath: 'meta::pure::metamodel::section::SectionIndex',
+  },
+  {
+    path: 'test::C',
+    content: {
+      _type: 'class',
+      name: 'C',
+      package: 'test',
+      properties: [
+        {
+          multiplicity: {
+            lowerBound: 1,
+            upperBound: 1,
+          },
+          name: 'z',
+          type: 'String',
+        },
+      ],
+      stereotypes: [
+        {
+          profile: 'temporal',
+          value: 'businesstemporal',
+        },
+      ],
+    },
+    classifierPath: 'meta::pure::metamodel::type::Class',
+  },
```

Comment:
lets add a a class that has `test::C` as a supertype class and assert on that as well.